app/lib/cds_importer/entity_mapper/base_mapper.rb

```diff
@@ -108,42 +108,21 @@ def parse
         if mapping_path.present?
           expanded.select! { |values| values.dig(*mapping_path.split(PATH_SEPARATOR)).present? }
         end
-        expanded.map { |values| create_instance(values) }
+        expanded.map(&method(:build_instance))
       end
 
       private
 
-      def create_instance(values)
-        normalized_values = normalized_values(mapped_values(values))
+      def build_instance(values)
+        values = mapped_values(values)
+        normalized_values = normalized_values(values)
         instance = entity_class.constantize.new
         instance.set_fields(normalized_values, entity_mapping.values)
       end
 
-      protected
-
-      def entity_class
-        self.class.entity_class.presence || raise(ArgumentError, "entity_class has not been defined: #{self.class}")
-      end
-
-      def mapping_path
-        self.class.mapping_path.presence
-      end
-
-      def entity_mapping
```

Comment:
Whilst the `delegate` approach is obviously much simpler, this will help future developers if they are getting odd behaviour whilst trying to implement new root entities?

Reply:
It's actually super difficult to get wrong now. You add a spec with your real xml node and check it parses cleanly/produces the resource you expect it too. I'll make it even more obvious with the specs coming for insertion/soft deletion, too